ehrql/dummy_data_nextgen/generator.py

```diff
@@ -187,42 +205,42 @@ def get_patient_column(self, column_name):
     def generate_patient_facts(self, patient_id):
         # Seed the random generator using the patient_id so we always generate the same
         # data for the same patient
-        self.rnd.seed(f"{self.random_seed}:{patient_id}")
-        # TODO: We could obviously generate more realistic age distributions than this
+        with self.seed(patient_id):
```

Comment:
Given that this method only has a single call site it might make sense to apply the context manager there and not have to indent the entire method body.

Reply:
I think in general I prefer the functions to be callable without having to do anything special at the callsite.

I did wonder about making this a decorator rather than a context manager, which would remove the indentation concern.

Reply:
Yes, that's fair enough. I'd leave it as is then.